python/pyspark/ml/torch/distributor.py

```diff
@@ -0,0 +1,179 @@
+                task_gpu_amount = int(self.sc.getConf().get(key))
+            else:
+                task_gpu_amount = 1 # for single node clusters
+            if task_gpu_amount < 1:
+                raise ValueError(
+                    f"The Spark conf `{key}` has a value "
+                    f"of {task_gpu_amount} but it "
+                    "should not have a value less than 1."
+                )
+            return math.ceil(self.num_processes / task_gpu_amount)
+        return self.num_processes
+    
+    def _validate_input_params(self):
+        if self.num_processes <= 0:
+            raise ValueError(f"num_proccesses has to be a positive integer")
+
+    def _check_encryption(self):
+        if "ssl_conf" not in self.__dict__:
+            raise NotImplementedError()
+        is_ssl_enabled = get_conf_boolean(
+            self.sc,
+            "spark.ssl.enabled",
+            "false"
+        )
+        ignore_ssl = get_conf_boolean(
+            self.sc,
+            self.ssl_conf,
+            "false"
+        )
+        if is_ssl_enabled:
+            name = self.__class__.__name__
+            if ignore_ssl:
+                warnings.warn(
+                    f"""
+                    This cluster has TLS encryption enabled;
+                    however, {name} does not
+                    support data encryption in transit.
+                    The Spark configuration
+                    '{self.ssl_conf}' has been set to
+                    'true' to override this
+                    configuration and use {name} anyway. Please
+                    note this will cause model
+                    parameters and possibly training data to
+                    be sent between nodes unencrypted.
+                    """,
+                    RuntimeWarning
+                )
+                return
+            raise Exception(
+                f"""
+                This cluster has TLS encryption enabled;
+                however, {name} does not support
+                data encryption in transit. To override
+                this configuration and use {name}
+                anyway, you may set '{self.ssl_conf}'
+                to 'true' in the Spark configuration. Please note this
+                will cause model parameters and possibly training
+                data to be sent between nodes unencrypted.
+                """
+            )
+
+class PyTorchDistributor(Distributor):
+    """
+    A class to support distributed training on PyTorch and PyTorch Lightning using PySpark.
+
+    .. versionadded:: 3.4.0
+
+    TODO: Add examples in the future
+    """
+    available_frameworks = ["pytorch", "pytorch-lightning"]
+    PICKLED_FUNC_FILE = "func.pickle"
+    TRAIN_FILE = "train.py"
+    
+    def __init__(self, framework: str, num_processes: int = 1, local_mode: bool = True, use_gpu: bool = True, spark: Optional[SparkSession] = None):
+        """Initializes the distributor
```

Comment:
Should better follow NumPy documentation style (https://numpydoc.readthedocs.io/en/latest/format.html)